crates/subspace-farmer/src/plot.rs

```diff
@@ -376,3 +304,201 @@ impl WeakPlot {
         self.inner.upgrade().map(|inner| Plot { inner })
     }
 }
+
+#[derive(Debug)]
+pub(crate) struct IndexToOffsetDB {
+    inner: DB,
+}
+
+impl From<DB> for IndexToOffsetDB {
+    fn from(inner: DB) -> Self {
+        Self { inner }
+    }
+}
+
+impl IndexToOffsetDB {
+    pub fn get(&self, key: u64) -> io::Result<Option<u64>> {
```

Comment:
The  key will be a hash of the piece index though. Also better to call it `piece_index_hash` if you decided to creeate an abstraction for it.

Reply:
I also find the `get` and `put` are kind of too abstract, some specific context could be nice. BTW, documenting what is the index and offset somewhere can be great, I only have a fuzzy idea about what they are.